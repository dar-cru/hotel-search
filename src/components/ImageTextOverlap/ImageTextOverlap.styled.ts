import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
  width: 242px;
  height: 150px;
  margin: 5px 10px 5px 0;
`;

export const ImageContainer = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
`;

export const OverlappingText = styled.p`
  position: absolute;
  background-color: white;
  padding: 6px;
  top: 12px;
  left: 0;
  box-sizing: border-box;
  color: red;
  margin: 0;
  font-size: 0.9em;
  font-weight: 500;
`;
