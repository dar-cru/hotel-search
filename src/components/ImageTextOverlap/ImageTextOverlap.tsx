import { Container, ImageContainer, OverlappingText } from './ImageTextOverlap.styled';

type ImageTextOverlapProps = {
  url: string;
  imgDesc: string;
  text: string;
};

const ImageTextOverlap = ({ url, imgDesc, text }: ImageTextOverlapProps) => {
  return (
    <Container>
      <ImageContainer src={url} alt={imgDesc} />
      <OverlappingText>{text}</OverlappingText>
    </Container>
  );
};

export default ImageTextOverlap;
