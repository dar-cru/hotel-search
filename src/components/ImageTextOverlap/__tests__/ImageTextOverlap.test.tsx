import { render, RenderResult } from '@testing-library/react';
import ImageTextOverlap from '../ImageTextOverlap';

describe('ImageTextOverlap component', () => {
  it('renders with correct styling and all values', () => {
    const tree: RenderResult = render(
      <ImageTextOverlap url="https://foo.com" imgDesc="An image of foo" text="Grab the latest deals!" />
    );

    const { container } = tree;
    expect(container).toMatchSnapshot();
  });
});
