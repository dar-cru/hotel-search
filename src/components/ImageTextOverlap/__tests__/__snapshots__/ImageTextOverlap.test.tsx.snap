// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageTextOverlap component renders with correct styling and all values 1`] = `
.c0 {
  position: relative;
  width: 242px;
  height: 150px;
  margin: 5px 10px 5px 0;
}

.c1 {
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  position: absolute;
  background-color: white;
  padding: 6px;
  top: 12px;
  left: 0;
  box-sizing: border-box;
  color: red;
  margin: 0;
  font-size: 0.9em;
  font-weight: 500;
}

<div>
  <div
    class="c0"
  >
    <img
      alt="An image of foo"
      class="c1"
      src="https://foo.com"
    />
    <p
      class="c2"
    >
      Grab the latest deals!
    </p>
  </div>
</div>
`;
